assets/app/scripts/services/data.js

```diff
@@ -856,8 +849,8 @@ angular.module('openshiftConsole')
       else if (eventData.type === "DELETED") {
         // Add this ourselves since the API doesn't add anything
         // this way the views can use it to trigger special behaviors
-        if (eventData.object.metadata) {
-          eventData.object.metadata.deleted = true;
+        if (eventData.object.metadata && !eventData.object.metadata.deletionTimestamp) {
```

Comment:
we have other stuff below it that would have blown up if eventData.object was undefined, but i can update them all in case we ever start doing watches that arent for resources